### PR TITLE
[WIP] elementsOf(Iterable) for contains.inAnyOrder.only #127

### PR DESCRIPTION
**Pull request by lelloman**
----
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/iterableContainsInAnyOrderOnlyCreators.kt [hunk 1]
```diff
@@ -89,3 +89,10 @@ fun <E : Any, T : Iterable<E?>> IterableContains.Builder<E?, T, InAnyOrderOnlySe
         assertionCreatorOrNull glue otherAssertionCreatorsOrNulls
     )
 )
+
+inline fun <reified E : Any, T : Iterable<E?>> IterableContains.Builder<E?, T, InAnyOrderOnlySearchBehaviour>.elementsOf(
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/iterableContainsInAnyOrderOnlyCreators.kt]:
```suggestion
inline fun <reified E : Any, T : Iterable<E>> IterableContains.Builder<E, T, InAnyOrderOnlySearchBehaviour>.elementsOf(
```

Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/iterableContainsInAnyOrderOnlyCreators.kt]: please add KDoc

**Comment by lelloman** [comment]
Hi @robstoll sorry again for the delay, my main problem is that I don't really understand the big picture. I saw the commit you referenced in the issue ([a0b539c](https://github.com/robstoll/atrium/commit/a0b539c0fb6434e3ba9beb9fb8d47393d32dc293)) but, given that I got the wip change here correctly, I don't understand how the 2 features are related

**Comment by robstoll** [comment]
Hi @lelloman your PR and https://github.com/robstoll/atrium/pull/157 are very similar. #157 was about adding `elementOf` to `contains.inAnyOrder.atLeast(1)` (see https://github.com/robstoll/atrium/pull/157/files#diff-9470ba377a3696d1eae2652a91019f8fR25) and your PR shall do the same for `contains.inAnyOrder.only`. Does that help to clarify things?

**Comment by lelloman** [comment]
Hi @robstoll, I still don't get it, [this function](https://github.com/robstoll/atrium/pull/157/files#diff-dfaa4b2cb89f389da137319d2d1511bdR108) looks like the one in my PR. I'm sorry but I can't wrap my head around this and I'll have to drop it :(

**Comment by robstoll** [comment]
The difference is the SearchBehavior. The function is using `InAnyOrderSearchBehaviour` and yours should use `InAnyOrderOnlySearchBehaviour` (notice the `Only`)

**Pull request closed without merging**